Service/OasService.php

```diff
@@ -74,6 +76,56 @@ public function createOas(): array
         return $oas;
     }//end createOas();
 
+    /**
+     * Adds the endpoints to an OAS Array.
+     *
+     * @param array $oas The OAS array where security should be added.
+     * @param Endpoint $endpoint The endpoint.
+     *
+     * @return array The OAS array including parameters for paths with id.
+     */
+    private function addParametersForPathWithId(array $oas, Endpoint $endpoint, array $pathArray): array
+    {
+        if (in_array('GET', $endpoint->getMethods()) === true
+            || in_array('PUT', $endpoint->getMethods()) === true
+            || in_array('PATCH', $endpoint->getMethods()) === true
+            || in_array('DELETE', $endpoint->getMethods()) === true
+        ) {
+            $oas['paths']['/' . implode('/', $pathArray) . '/{id}']['parameters'][] = [
+                'name' => 'id',
+                'in' => 'path',
+                'description' => 'Unieke resource identifier (UUID4)',
+                'required' => true,
+                'schema' => [
+                    'type' => 'string',
+                    'format' => 'uuid'
+                ],
+            ];
+        }//end if
+
+        return $oas;
+    }
+
+    /**
+     * Adds the endpoints to an OAS Array.
+     *
+     * @param Endpoint $endpoint The endpoint.
+     *
+     * @return array The path array
+     */
+    private function getPathArray(Endpoint $endpoint): array
+    {
+        $pathArray = [];
+        foreach ($endpoint->getPath() as $path) {
+            if ($path === 'id') {
+                continue;
+            }
+            $pathArray[] = $path;
+        }//end foreach
```

Comment:
Some endpoints have multiple 'id' keys in the path array. (/entitynames/{id}/subentitynames/{id})
If you only want to remove the last 'id' key from the array this might be cleaner and faster:
$pathArray = $endpoint->getPath();
if (end($pathArray) === 'id') {
            array_pop($pathArray);
        }